Upgrade packages that use old versions of graceful-fs

thomasjo opened the pull request:
Bumps packages that have dependencies on graceful-fs to versions that uses graceful-fs v4 or higher. 

This is **necessary** for Node.js v6 support.

TODO:
- [x] Bump fstream.
- [x] Open PRs against [less-cache](https://github.com/atom/less-cache) — https://github.com/atom/less-cache/pull/10
  - [x] Bump less-cache once PRs have shipped.
- [ ] Fix regression during the `prebuild-less` task. See https://github.com/atom/atom/pull/12176#issuecomment-240339043

thomasjo committed:
:arrow_up: fstream

Bump to a version that uses graceful-fs v4. Needed for Node v6 support

thomasjo committed:
:arrow_up: less-cache@0.24.0

thomasjo commented:
@atom/feedback I think this should be good to go. Would appreciate some 👀 and maybe a bit of 🚬 testing.

thomasjo commented:
Nevermind, turns out this causes a problem during `script/build` (and possibly elsewhere)

```
Running "prebuild-less:src" (prebuild-less) task
Warning: variable @fold is undefined Use --force to continue.

Aborted due to warnings.
```

maxbrunsfeld commented:
> this causes a problem during `script/build`

I've tracked this problem down to what I believe is a bug in `less`. I have opened a PR fixing the bug: https://github.com/less/less.js/pull/2953

maxbrunsfeld committed:
Merge branch 'master' into tj-upgrade-deprecated-packages

maxbrunsfeld committed:
Avoid using less imports in nested scopes

Apparently, these imports of 'octicon-utf-codes.less' within mixin
definitions cause any subsequent imports of this file to become noops,
because of the way less dedupes imports. The result is that the variables
defined in that file are only available in the nested scope.

This didn't happen in older versions of less because of bugs which
have been fixed since less 2.7.

Signed-off-by: Antonio Scandurra <as-cii@github.com>

maxbrunsfeld commented on static/variables/octicon-mixins.less:
```diff
@@ -1,3 +1,5 @@
+@import "octicon-utf-codes.less";
+
```
See https://github.com/less/less.js/pull/2953

maxbrunsfeld commented:
Interesting; the janky build passed, but the circle build failed with

```
Expected '#header {
  color: #4D926F;
}
h2 {
  color: #4D926F;
}
' to be '#header {
  color: #4d926f;
}
h2 {
  color: #4d926f;
}
'
```

Maybe the new version of `less` normalizes color literals to always use capital letters?

maxbrunsfeld committed:
Don't expect color variables to be lowercased in ThemeManager spec

maxbrunsfeld commented:
OK, it turns out that the less bug I mentioned above is the expected behavior, so I've changed our base stylesheets to avoid imports in nested scopes. Now I've run into another less bug:

https://github.com/less/less.js/pull/2955
